dhis2/dhis2-core
# feat: add IdObjectDeletionHandler for users [DHIS2-13470]

https://dhis2.atlassian.net/browse/DHIS2-13470
- When deleting a User the `IdObjectDeletionHandler` will check if that User is  linked to any IdentifiableObject  of all Metadata object types. 
- The look up query use property `createdBy` and `lastUpdatedBy`. However, not all objects which extends `BaseIdentifiableObject` have both properties. So we need to use the schema to check before execute the loop up query.
- Added test in `UserServiceTest`

#### Manual testing
1. Create a new UserA, login with that user and create a DataElement. 
2. Login with system user and delete UserA
3. Expected: error is shown with message indicating UserA is linked to a DataElement.
